src/main/java/practice/StreamPractice.java

```diff
@@ -75,6 +101,10 @@ public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
      * parametrized with Candidate in CandidateValidator.
      */
     public List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return candidates.stream()
+                .filter(c -> new CandidateValidator().test(c))
```

Comment:
not fixed, our Candidate validator is a Predicate itself so we can have it like this
```suggestion
                .filter(new CandidateValidator())
```